src/sources/mongodb_metrics/mod.rs

```diff
@@ -1099,53 +1105,60 @@ mod integration_tests {
     }
 
     async fn test_instance(endpoint: String) {
-        let host = ClientOptions::parse(endpoint.as_str()).await.unwrap().hosts[0].to_string();
-        let namespace = "vector_mongodb";
-
-        let (sender, mut recv) = SourceSender::new_test();
-
-        let endpoints = vec![endpoint.clone()];
-        tokio::spawn(async move {
-            MongoDbMetricsConfig {
-                endpoints,
-                scrape_interval_secs: 15,
-                namespace: namespace.to_owned(),
+        assert_source_compliance(&SOCKET_PULL_SOURCE_TAGS, async {
+            let host = ClientOptions::parse(endpoint.as_str()).await.unwrap().hosts[0].to_string();
+            let namespace = "vector_mongodb";
+
+            let (sender, mut recv) = SourceSender::new_test();
+
+            let endpoints = vec![endpoint.clone()];
+            tokio::spawn(async move {
+                MongoDbMetricsConfig {
+                    endpoints,
+                    scrape_interval_secs: 15,
+                    namespace: namespace.to_owned(),
+                }
+                .build(SourceContext::new_test(sender, None))
+                .await
+                .unwrap()
+                .await
+                .unwrap()
+            });
+
+            // TODO: We should have a simpler/cleaner method for this sort of collection, where we're essentially waiting
+            // for a burst of events, and want to debounce ourselves in terms of stopping collection once all events in the
+            // burst have been collected. This code here isn't bad or anything... I've just noticed now that we do it in a
+            // few places, and we could solve it in a cleaner way, most likely.
+            let event = timeout(Duration::from_secs(30), recv.next())
```

Comment:
Not your change, but... 30 seconds seems like quite the long timeout.

Reply:
Yeah, it seems predicated on the 15 second scrape interval... but it's not clear if the scrape interval can't simply be lowered down to one second, for example.